compliance/collection/main.go

```diff
@@ -182,17 +217,106 @@ func manageNodeScanLoop(ctx context.Context, rescanInterval time.Duration, scann
 	return sensorC
 }
 
-func scanNode(nodeName string, scanner nodeinventorizer.NodeInventorizer) (*sensor.MsgFromCompliance, error) {
-	result, err := scanner.Scan(nodeName)
+func runCachedScan(opts *cachedScanOpts) (*storage.NodeInventory, error) {
+	inventory, err := opts.scanner.Scan(opts.nodeName)
 	if err != nil {
 		return nil, err
 	}
+	inv, err := proto.Marshal(inventory)
+	if err != nil {
+		return nil, err
+	}
+	if err := os.WriteFile(fmt.Sprintf("%s/last_scan", opts.inventoryCachePath), inv, 0600); err != nil {
```

Comment:
[nit] This `fmt.Sprintf("%s/last_scan", opts.inventoryCachePath)` is done multiple times as well as `fmt.Sprintf("%s/backoff", opts.inventoryCachePath)`. Do you mind just doing each only once and storing in corresponding new `opts` fields?

Reply:
Thank you, I have moved this into the options struct